### PR TITLE
Refactor/use web crypto

### PR DESCRIPTION
**Pull request by javadkh2**

### CODE REVIEW

**Review by alber70g — APPROVED** [review]

File: packages/libs/hd-wallet/src/SLIP10/kadenaGenKeypairFromSeed.ts [hunk 1]
```diff
@@ -48,17 +56,17 @@ export function kadenaGenKeypairFromSeed(
...
   indexOrRange: number | [number, number],
   derivationPathTemplate: string = `m'/44'/626'/<index>'`,
-): [string, EncryptedString] | Array<[string, EncryptedString]> {
+): Promise<[string, EncryptedString] | Array<[string, EncryptedString]>> {
   if (typeof seed !== 'string' || seed === '') {
     throw new Error('No seed provided.');
   }
 
-  const seedBuffer = kadenaDecrypt(password, seed);
+  const seedBuffer = new Uint8Array(await kadenaDecrypt(password, seed));
```
Comment (barthuijgen) [inline packages/libs/hd-wallet/src/SLIP10/kadenaGenKeypairFromSeed.ts:69]: doesn't kadenaDecrypt already return a Uint8Array?

File: packages/libs/hd-wallet/src/SLIP10/utils/sign.ts [hunk 2]
```diff
@@ -10,15 +10,18 @@ export interface ISignatureWithPublicKey {
...
 /**
  * Derive a key pair using a seed and an index.
- * @param {Uint8Array} seed - The seed for key derivation.
+ * @param {ArrayBuffer} seed - The seed for key derivation.
  * @param {number} index - The index for key derivation.
  * @returns {{ privateKey: string; publicKey: string }} - Returns the derived private and public keys.
  */
 export const deriveKeyPair = (
-  seed: Uint8Array,
+  seed: ArrayBuffer,
```
Comment (barthuijgen) [inline packages/libs/hd-wallet/src/SLIP10/utils/sign.ts:18]: Why are you moving from Uint8Array to ArrayBuffer?

Comment (javadkh2) [reply packages/libs/hd-wallet/src/SLIP10/utils/sign.ts:18]: I didn't move from Uint8Array, it is just a view on to of ArrayBuffer and actually it just extends the class.
So you can also pass Uint8Array or even Buffer

File: packages/libs/hd-wallet/src/SLIP10/kadenaGenKeypairFromSeed.ts (hunk 1 above)
Comment (javadkh2) [reply packages/libs/hd-wallet/src/SLIP10/kadenaGenKeypairFromSeed.ts:69]: Yes I will fix this

**Review by barthuijgen — APPROVED** [review]

**Pull request merged**